### ISSUE TITLE
feat(depute): gestion des photos

### ISSUE THREAD

**Issue by revolunet** [regardscitoyens/NosDeputes2-front#4]
A propos des photos des députés :

Il y a sur la version 1 pas mal de [traitements "runtime" des images](https://github.com/regardscitoyens/nosdeputes.fr/blob/master/apps/frontend/modules/parlementaire/templates/photoSuccess.php) :
 - une notion de "autoflip" qui est un flag dans la table "parlementaires"... je n'ai pas vraiment compris a quoi sert ce flag. peut-être @teymour s'en souvient ?
 - une pastille de couleur ajoutée selon le groupe
 - possibilité de resize/flip via des paramètres d'url
 
Une possibilité serait, au niveau du repo "data", qu'on s'occupe de les préprocesser en amont pour pouvoir les exploiter de manière statique.

**Comment by implicitdef** [comment]
Autoflip c'était pour que les photos "regardent" du même côté. Par défaut sur certaines le député était légèrement vers la droite ou vers la gauche, les anciens voulaient qu'il n'y ait aucun "biais", et donc marquer les photos qu'il fallait flipper (je pense qu'on peut faire sans)

La pastille de couleur ça on peut tout refaire en CSS aucun souci

Je sais pas trop quel est l'approche à suivre. Le plus simple pour démarrer serait de mettre les photos directement dans le dossiers public de Next.js, comme des assets normaux. Le souci c'est qu'on sera pas à jour quand un nouveau député arrive.

Peut-être on pourrait faire une moulinette qui tous les jours va récupérer les dernières images depuis le site de l'assemblée (ou leur open data si c'est dedans ?) et les poser sur un truc type S3 (peut-être avec plusieurs dimensions si besoin) d'où le front pourra les servir directement

**Comment by revolunet** [comment]
Ok pour faire simple pour démarrer.

On pourrait faire une commande yarn lancée par une GH action qui va récupérer+traiter (resize) régulièrement les photos depuis le site de l'AN et les commit sur le repo dans `/public` en cas de changement. je peux proposer une PR ca si ca te parait cohérent.

le truc de la pastille de couleur en CSS oui... il faut "juste" gérer les éventuels différentes tailles, responsive... ca peut faire un composant react sympa :)

**Comment by revolunet** [comment]
(Le flip peut aussi être géré en CSS en fait)

**Comment by implicitdef** [comment]
Yes pour la GH action ça me parait un bon début, tu peux y aller

**Comment by revolunet** [comment]
je pense qu'on peut close, il ne reste a priori plus que la gestion de la pastille du groupe

**Comment by RouxRC** [comment]
Hello, je trouvais assez cool personnellement que les gens qui réutilisent l'API de ND présentent du coup sur leurs sites également des photos avec nos pastilles maison (ce qui permettait aussi très rapidement de les identifier), du coup je plaiderai personnellement en faveur de rajouter les pastilles en dur dans votre GH action plutôt qu'en css sur le site

**Comment by RouxRC** [comment]
(le flip j'aurai tendance à l'oublier par contre ;) )

**Comment by revolunet** [comment]
Je pense qu'on peut l'ajouter sur le JPG directement, d'autant que l'info de la "couleur" est fournie directement sur le site de l'AN : https://www2.assemblee-nationale.fr/deputes/liste/photo

@RouxRC est-ce que NosDeputes utilise les memes couleurs que celles de l'AN ?

<img width="728" alt="Capture d’écran 2022-11-06 à 16 09 39" src="https://user-images.githubusercontent.com/124937/200178769-33ed2e36-2ade-4e93-bb9d-d8bb5164a7a1.png">

Je peux tenter d'ajouter ca au moment du fetch si ok pour @implicitdef

**Comment by RouxRC** [comment]
ha non je découvre ces couleurs sur le site de l'AN je ne les avais jamais vues encore.
Non jusque là les couleurs étaient l'un de nos rares choix éditoriaux, réglé en configuration ici : https://github.com/regardscitoyens/nosdeputes.fr/blob/master/config/app.yml.example#L14-L32
(et je les trouve plus justes, j'ai un peu de mal avec le violet pour renaissance côté an je dois avouer...)

**Comment by implicitdef** [comment]
En fait je ne suis pas très convaincu non plus par ces pastilles en termes d'UX. Je trouve difficile de mémoriser les couleurs, et je n'étais même pas sûr que ça représentait le groupe.
Je pense que c'est quelque chose qu'il faut voir au moment du design de la page, suivant le contexte, est-ce qu'on veut afficher juste une couleur, ou rajouter l'acronyme du groupe, est-ce qu'on veut que ce soit sur l'image, ou à côté, et dans quelles dimensions ? car cela mérite peut-être de changer si l'image fait telle ou telle taille

Bref tu peux le faire si tu veux mais suivant ce qu'on fait niveau design plus tard je ne suis pas sûr que ça restera

**Comment by RouxRC** [comment]
C'est à vous d'en débattre entre vous clairement. L'argument principal à mon avis de la présence sur les photos, c'est de pouvoir identifier le ou les groupes d'appartenance des députés lorsqu'ils apparaissent dans d'autres contextes, par exemple dans un compte-rendu de débat, dans une liste de signataires d'amendement ou de propositions de lois, etc.

**Comment by revolunet** [comment]
ok on met ce sujet à débattre dans la catégorie ux/design :)
